lib/ansible/inventory/manager.py

```diff
@@ -367,7 +367,7 @@ def get_hosts(self, pattern="all", ignore_limits=False, ignore_restrictions=Fals
                 from random import shuffle
                 shuffle(hosts)
             elif order not in [None, 'inventory']:
-                AnsibleOptionsError("Invalid 'order' specified for inventory hosts: %s" % order)
+                raise AnsibleError("Invalid 'order' specified for inventory hosts: %s" % order)
```

Comment:
I'm actually worried why are we 'silently ignoring' options exceptions, that should not be the case, the 'type' is correct in this case, but something is ignoring it up the exception handling chain.

Reply:
@bcoca 'silently ignoring' options exceptions doesn't seem to be happening. But there isn't a CLI flag for order, so the output from the exception doesn't provide anything helpful to the user.
Should there be a CLI option? Or is adding the permitted values to the error fine?
```
-                raise AnsibleError("Invalid 'order' specified for inventory hosts: %s" % order)
+                raise AnsibleOptionsError("Invalid 'order' specified for inventory hosts: %s. 'order' may be one "
+                                          "of [null|inventory|reverse_inventory|reverse_sorted|shuffle|sorted]." % order)
```

Reply:
I might have been using 'options' to mean more than CLI options .. this might be better classified as a ParserError? or do we need a KeywordError?